### DISCUSSION TITLE
HW 1 Part 3

### DISCUSSION THREAD

**Discussion by chrischaplinsky3** [cooperrc/computational-mechanics#141] [Q&A]
I'm pretty lost on part 3, I can't seem to get the data loaded properly, causing errors in my code. Can anyone help with this? I'm not sure how to move forward with it.

**Comment by C-Doll** [comment]
I ended up finding the source of the information here: https://raw.githubusercontent.com/*YOURUSERNAME*/computational-mechanics/master/data/world_population_1900-2020.csv

And I just used that link and replaced it with what he had listed and it worked. 
"year, pop = np.loadtxt('../data/world_population_1900-2020.csv',skiprows=1,delimiter=',',unpack=True)"

"year, pop = np.loadtxt('https://raw.githubusercontent.com/*YOURUSERNAME*/computational-mechanics/master/data/world_population_1900-2020.csv',skiprows=1,delimiter=',',unpack=True)"

But I think the proper way to read the data without doing all that is to follow the steps from his most recent youtube video.
https://youtu.be/5D_r_jzoLls?t=311 at that time stamp.
It shows you how to clone the information into the "work" section of the jupiter notebook. I am not 100% sure why the step is necessary, but it works.

Hope that helps.

**Comment by cooperrc** [comment]
Usually this is an issue of files being in the wrong places. 

if this happens, 

```python
import numpy as np
year, pop = np.loadtxt('../data/world_population_1900-2020.csv',skiprows=1,delimiter=',',unpack=True)
print('years=',year)
print('population =', pop)

---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
/tmp/ipykernel_90/2152235788.py in <module>
      1 import numpy as np
----> 2 year, pop = np.loadtxt('./data/world_population_1900-2020.csv',skiprows=1,delimiter=',',unpack=True)
      3 print('years=',year)
      4 print('population =', pop)

/opt/conda/lib/python3.9/site-packages/numpy/lib/npyio.py in loadtxt(fname, dtype, comments, delimiter, converters, skiprows, usecols, unpack, ndmin, encoding, max_rows, like)
   1063             fname = os_fspath(fname)
   1064         if _is_string_like(fname):
-> 1065             fh = np.lib._datasource.open(fname, 'rt', encoding=encoding)
   1066             fencoding = getattr(fh, 'encoding', 'latin1')
   1067             fh = iter(fh)

/opt/conda/lib/python3.9/site-packages/numpy/lib/_datasource.py in open(path, mode, destpath, encoding, newline)
    192 
    193     ds = DataSource(destpath)
--> 194     return ds.open(path, mode, encoding=encoding, newline=newline)
    195 
    196 

/opt/conda/lib/python3.9/site-packages/numpy/lib/_datasource.py in open(self, path, mode, encoding, newline)
    529                                       encoding=encoding, newline=newline)
    530         else:
--> 531             raise IOError("%s not found." % path)
    532 
    533 

OSError: ./data/world_population_1900-2020.csv not found.
```

Then, the notebook is probably in a different place. 

try this instead, 

```python
import numpy as np
year, pop = np.loadtxt('/home/jovyan/work/data/world_population_1900-2020.csv',skiprows=1,delimiter=',',unpack=True)
print('years=',year)
print('population =', pop)
```
